pkg/utils/gardener/gardenlet/gardenlet_test.go

```diff
@@ -80,18 +80,20 @@ var _ = Describe("Gardenlet", func() {
 		})
 
 		It("should return that the seed is a self-hosted shoot", func() {
-			Expect(fakeClient.Create(ctx, &appsv1.Deployment{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "gardenlet",
-					Namespace: "kube-system",
-				},
-			})).To(Succeed())
+			Expect(fakeClient.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "kube-system", Labels: map[string]string{"gardener.cloud/role": "shoot"}}})).To(Succeed())
 			Expect(SeedIsSelfHostedShoot(ctx, fakeClient)).To(BeTrue())
 		})
 
-		It("should return that the seed is not a self-hosted shoot because no gardenlet deployment found", func() {
+		It("should return that the seed is not a self-hosted shoot because kube-system namespace is not labeled", func() {
```

Comment:
Can you add another test case for kube-system in hosted shoots. Here, we have the `gardener.cloud/purpose=kube-system` label